src/backend/bigquery/buildSqlQuery.ts

```diff
@@ -150,13 +150,28 @@ export const buildSqlQuery = function (
   let linkClause = ''
   if (searchParams.linkSearchUrl !== '') {
     // Link search: look for url as substring
-    linkClause = `
-      AND EXISTS
-        (
-          SELECT 1 FROM UNNEST (hyperlinks) AS link
-          WHERE CONTAINS_SUBSTR(link.link_url, @link)
-        )
-    `
+    if (searchParams.linksExactMatch) {
+      linkClause = `
+        AND EXISTS
+          (
+            SELECT 1 FROM UNNEST (hyperlinks) AS link
+            WHERE
+            @link = link.link_url
+            OR CONCAT('http://www.gov.uk', @link) = link.link_url
+            OR CONCAT('http://www.gov.uk', @link) = link.link_url
+            OR CONCAT('https://www.gov.uk', @link) = link.link_url
+            OR CONCAT('https://www.gov.uk', @link) = link.link_url
```

Comment:
Two of these clauses seem to be duplicates of the other two.

Reply:
I'm uneasy about automatically allowing `https://ww.gov.uk` in an "exact" match. Do we know that any users expect it?

If we do this, it should also be reflected in the search description text, e.g. "25 results for pages that link to /foo" should be "25 results for pages that link to https://www.gov.uk/brexit".

Reply:
Good point. I don't think I thought carefully enough when I wrote that issue, and I don't think we have any user research findings about it. Either way, the search description text should reflect what is actually being searched for.